components/menu-dropdown/menu-dropdown.jsx

```diff
@@ -615,6 +619,7 @@ class MenuDropdown extends React.Component {
 
 		if (!isOpen && this.props.openOn === 'hover') {
 			this.handleOpen();
+			this.setFocus();
```

Comment:
I noticed this is triggering 2 calls to `props.onOpen`. One is coming from `handleOpen` and the other one from `setFocus` which indirectly calls `handleFocus`. Might want to try this instead : 
```
if (this.props.openOn === 'hover') {
    this.setFocus();
} else if (!isOpen) {
    this.setFocus();
}
```

Reply:
I am looking into this.